Patches OS X detection out of os-prober hook on non-x86 architectures. The
menu entries generated for those are invalid for non-x86 Mac stuff.
--- a/util/grub.d/30_os-prober.in
+++ b/util/grub.d/30_os-prober.in
@@ -42,6 +42,11 @@ if [ -z "${OSPROBED}" ] ; then
 fi
 
 osx_entry() {
+    # GRUB won't load OS X outside of x86, no entry
+    case "x`uname -m`" in
+        xi?86|xx86_64) ;;
+        *) return ;;
+    esac
     if [ x$2 = x32 ]; then
         # TRANSLATORS: it refers to kernel architecture (32-bit)
 	bitstr="$(gettext "(32-bit)")"
